packages/thirdweb/src/react/core/utils/wallet.ts

```diff
@@ -100,9 +103,67 @@
     ensName: ensNameQuery.data,
   });
 
-  const socialProfileQuery = useSocialProfiles({
-    client,
-    address: activeAccount?.address,
+  const connectedWallets = useConnectedWallets();
+
+  const { data: linkedAccounts } = useProfiles();
+
+  const socialProfileQuery = useQuery({
+    queryKey: [
+      "social-profiles",
+      activeAccount?.address,
+      linkedAccounts,
+      connectedWallets,
+    ],
+    queryFn: async () => {
+      if (!activeAccount?.address) {
+        return [];
+      }
+
+      const primaryAccountProfiles = await getSocialProfiles({
+        client,
+        address: activeAccount?.address,
+      });
+
+      const connectedWalletProfiles = await Promise.all(
+        connectedWallets
+          .map((w: Wallet) => w.getAccount())
+          .filter(
+            (a: Account | undefined) =>
+              a !== undefined && a.address !== activeAccount.address,
+          )
+          .flatMap(async (account) => {
+            return await getSocialProfiles({
+              client,
+              address: (account as Account).address,
+            });
+          }),
+      );
```

Comment:
I don't think if I'm connected with `walletA` and `walletB` I want social profiles from `walletB` to show up as my connected wallet?

say walletA has nothing, walletB has ens

now we'd show `walletB` ens as the connected address? That's simply wrong, no?